# Unused analyses

In an effort to gauge the performance of differential expression software, multiple differential expression pipelines were performed on these data. In the end, we decided to use featureCounts because of it's speed, and edgeR exact test because of it's superior performance with low replicate RNA-seq studies (Schurch et al. 2016, doi:  10.1261/rna.053959.115). This folder contains the code and output for analyses that were not used. In general, DESEQ coroborated edgeR results. Ballgown produced vastly different results, with almost no differentially expressed genes returned. It was difficult to gauge why this was the case, as the algorithms used by StringTie and Ballgown are somewhat opaque (which, in the end, was an additional reason we opted not to use this approach, as it was difficult to understand the underlying assumptions of the model). Limma produced fewer results than the edgeR exact test, which we interpretted as indicating that limma detected fewer true differentially expressed genes while maintaining a near equal rate of false positives (see Schurch et al., 2016). 
